model_compression_toolkit/target_platform_capabilities/tpc_io_handler.py

```diff
@@ -15,12 +15,34 @@
 from pathlib import Path
 from typing import Union
 
-from model_compression_toolkit.logger import Logger
-from model_compression_toolkit.target_platform_capabilities.schema.mct_current_schema import TargetPlatformCapabilities
-import json
+import model_compression_toolkit.target_platform_capabilities.schema.mct_current_schema as schema
+from model_compression_toolkit.target_platform_capabilities.schema.schema_compatability import is_tpc_instance, \
+    tpc_to_current_schema_version
 
 
-def load_target_platform_capabilities(tpc_obj_or_path: Union[TargetPlatformCapabilities, str]) -> TargetPlatformCapabilities:
+def _get_tpc_from_json(tpc_path):
+    path = Path(tpc_path)
+
+    if not path.exists() or not path.is_file():
+        raise FileNotFoundError(f"The path '{tpc_path}' is not a valid file.")
+    # Verify that the file has a .json extension
+    if path.suffix.lower() != '.json':
+        raise ValueError(f"The file '{path}' does not have a '.json' extension.")
+    try:
+        with path.open('r', encoding='utf-8') as file:
+            data = file.read()
+    except OSError as e:
+        raise ValueError(f"Error reading the file '{tpc_path}': {e.strerror}.") from e
+
+    try:
+        return schema.TargetPlatformCapabilities.parse_raw(data)
+    except ValueError as e:
+        raise ValueError(f"Invalid JSON for loading TargetPlatformCapabilities in '{tpc_path}': {e}.") from e
+    except Exception as e:
+        raise ValueError(f"Unexpected error while initializing TargetPlatformCapabilities: {e}.") from e
+
+
+def load_target_platform_capabilities(tpc_obj_or_path: Union[schema.TargetPlatformCapabilities, str]) -> schema.TargetPlatformCapabilities:
```

Comment:
type of `tpc_obj_or_path` is not necessarily `schema.TargetPlatformCapabilities` since this is before possible conversion